pyteal/ast/router.py

```diff
@@ -0,0 +1,547 @@
+                )
+            else:
+                output_temp: abi.BaseType = cast(
+                    OutputKwArgInfo, handler.output_kwarg_info
+                ).abi_type.new_instance()
+                subroutine_call: abi.ReturnedValue = cast(
+                    abi.ReturnedValue, handler(*arg_abi_vars)
+                )
+                return Seq(
+                    *decode_instructions,
+                    subroutine_call.store_into(output_temp),
+                    abi.MethodReturn(output_temp),
+                    Approve(),
+                )
+
+    def __add_bare_call(self, oc_actions: OCActions) -> None:
+        if oc_actions.is_empty():
+            raise TealInputError("the OnCompleteActions is empty.")
+        bare_app_calls: dict[EnumInt, OCAction] = oc_actions.dictify()
+
+        cs_calls = bare_app_calls[OnComplete.ClearState]
+        if cs_calls.on_call is not None:
+            on_call = cast(
+                Expr | SubroutineFnWrapper | ABIReturnSubroutine,
+                cs_calls.on_call,
+            )
+            wrapped = Router.wrap_handler(False, on_call)
+            self.categorized_clear_state_ast.bare_calls.append(
+                CondNode(Int(1), wrapped)
+            )
+        if cs_calls.on_create is not None:
+            on_create = cast(
+                Expr | SubroutineFnWrapper | ABIReturnSubroutine,
+                cs_calls.on_create,
+            )
+            wrapped = Router.wrap_handler(False, on_create)
+            self.categorized_clear_state_ast.bare_calls_create.append(
+                CondNode(Txn.application_id() == Int(0), wrapped)
+            )
+
+        approval_calls = {
+            oc: oc_action
+            for oc, oc_action in bare_app_calls.items()
+            if str(oc) != str(OnComplete.ClearState)
+        }
+
+        for oc, approval_bac in approval_calls.items():
+            if approval_bac.on_call:
+                on_call = cast(
+                    Expr | SubroutineFnWrapper | ABIReturnSubroutine,
+                    approval_bac.on_call,
+                )
+                wrapped = Router.wrap_handler(False, on_call)
+                self.categorized_approval_ast.bare_calls.append(
+                    CondNode(Txn.on_completion() == oc, wrapped)
+                )
+            if approval_bac.on_create:
+                on_create = cast(
+                    Expr | SubroutineFnWrapper | ABIReturnSubroutine,
+                    approval_bac.on_create,
+                )
+                wrapped = Router.wrap_handler(False, on_create)
+                self.categorized_approval_ast.bare_calls_create.append(
+                    CondNode(
+                        And(Txn.application_id() == Int(0), Txn.on_completion() == oc),
+                        wrapped,
+                    )
+                )
+
+    def add_method_handler(
+        self,
+        method_call: ABIReturnSubroutine,
+        method_overload_name: str = None,
```

Comment:
@ahangsu Minor:  I _think_ _override_ rather than _overload_ is the intended term here.  As I understand it, the user is overriding name from `method_call` with the supplied argument.  Does that match your understanding?

If you agree, there's a few places where the naming convention appears.

Reply:
👍

Reply:
ugh my english vocabulary... 🤦